backend/routes/appointments/viewAppointmentsRouter.js

```diff
@@ -33,16 +33,74 @@ module.exports = ({ prisonApi, whereaboutsApi, logError }) => async (req, res) =
     whereaboutsApi.searchGroups(res.locals, agencyId),
   ])
 
-  const videoLinkAppointmentIds = appointments
-    .filter(appointment => appointment.appointmentTypeCode === 'VLB')
-    .map(videoLinkAppointment => videoLinkAppointment.id)
+  const videoLinkAppointmentsWithTimes = appointments.filter(appointment => appointment.appointmentTypeCode === 'VLB')
+  const videoLinkAppointmentIds = videoLinkAppointmentsWithTimes.map(videoLinkAppointment => videoLinkAppointment.id)
 
   const videoLinkAppointmentResponse = await whereaboutsApi.getVideoLinkAppointments(
     res.locals,
     videoLinkAppointmentIds
   )
+
   const videoLinkAppointments = (videoLinkAppointmentResponse && videoLinkAppointmentResponse.appointments) || []
   const videoLinkAppointmentsMadeByTheCourt = videoLinkAppointments.filter(appointment => appointment.madeByTheCourt)
+  const videoLinkAppointmentsWithTimesAndBookingIds = videoLinkAppointmentsWithTimes.map(appointment => {
```

Comment:
Can we move all this to a function (or 2) - it makes the function hard to follow.

I have a suggestion - can we not do all this as part of a mapping function when making `appointmentsEnhanced` below. It would be done after the filtering and would look for VLBs. If one was found then it would call a function to calculate the `appointmentTimes`. Not sure how easy it would be though - we can go through it together to see if it is feasible or better.